framework/src/context_vulkan.cpp

```diff
@@ -872,156 +872,53 @@ namespace gvk
-			avk::attachment::declare_for(const_referenced(aWindow->mSwapChainImageViews[0]), avk::on_load::clear, avk::color(0), avk::on_store::store)
-		};
-		auto additionalAttachments = aWindow->get_additional_back_buffer_attachments();
-		renderpassAttachments.insert(std::end(renderpassAttachments), std::begin(additionalAttachments), std::end(additionalAttachments));
-		aWindow->mBackBufferRenderpass = create_renderpass(renderpassAttachments);
-		aWindow->mBackBufferRenderpass.enable_shared_ownership(); // Also shared ownership on this one... because... why noooot?
-
-		// Create a back buffer per image
-		aWindow->mBackBuffers.reserve(imagesInFlight);
-		for (auto& imView: aWindow->mSwapChainImageViews) {
-			auto imExtent = imView->get_image().config().extent;
-
-			// Create one image view per attachment
-			std::vector<avk::resource_ownership<avk::image_view_t>> imageViews;
-			imageViews.reserve(renderpassAttachments.size());
-			imageViews.push_back(avk::shared(imView)); // The color attachment is added in any case
-			for (auto& aa : additionalAttachments) {
-				if (aa.is_used_as_depth_stencil_attachment()) {
-					auto depthView = create_depth_image_view(avk::owned(create_image(imExtent.width, imExtent.height, aa.format(), 1, avk::memory_usage::device, avk::image_usage::read_only_depth_stencil_attachment))); // TODO: read_only_* or better general_*?
-					imageViews.emplace_back(std::move(depthView));
-				}
-				else {
-					imageViews.emplace_back(create_image_view(avk::owned(create_image(imExtent.width, imExtent.height, aa.format(), 1, avk::memory_usage::device, avk::image_usage::general_color_attachment))));
-				}
-			}
-
-			aWindow->mBackBuffers.push_back(create_framebuffer(avk::shared(aWindow->mBackBufferRenderpass), std::move(imageViews), imExtent.width, imExtent.height));
-			aWindow->mBackBuffers.back().enable_shared_ownership();
-		}
-		assert(aWindow->mBackBuffers.size() == imagesInFlight);
-
-		// Transfer the backbuffer images into a at least somewhat useful layout for a start:
-		for (auto& bb : aWindow->mBackBuffers) {
-			const auto n = bb->image_views().size();
-			assert(n == aWindow->get_renderpass()->number_of_attachment_descriptions());
-			for (size_t i = 0; i < n; ++i) {
-				bb->image_view_at(i)->get_image().transition_to_layout(aWindow->get_renderpass()->attachment_descriptions()[i].finalLayout, avk::sync::wait_idle(true));
-			}
-		}
+		auto newSwapChain = device().createSwapchainKHRUnique(aWindow->mSwapChainCreateInfo);
+		
+		//context().device().waitIdle(); // TODO: Necessary?
+		auto lifetime_handler = [aWindow](vk::UniqueSwapchainKHR&& rhs) { aWindow->handle_lifetime(std::move(rhs)); };
 
-		// ============= SYNCHRONIZATION OBJECTS ===========
-		// per IMAGE:
+		// assign the new swap chain instead of the old one, if one exists
+		avk::swap_and_dispose_rhs(newSwapChain, std::move(aWindow->mSwapChain), lifetime_handler);			
+		
+		aWindow->construct_backbuffers(onlyUpdate);
+		
+		if (!onlyUpdate)
 		{
```

Comment:
```suggestion
		if (!onlyUpdate) {
```

and, use `enum struct` if an `enum struct` is being introduced as requested above!